dhall/src/Dhall/Syntax.hs

```diff
@@ -314,6 +318,37 @@ instance Bifunctor RecordField where
         RecordField (k <$> s0) (first k value)
     second = fmap
 
+{-| Record the label of a function or a function-type expression
+
+For example,
+
+> λ({- A -} a {- B -} : {- C -} T) -> e
+
+will be instantiated as follows:
+* @fbSrc0@ corresponds to the @A@ comment
+* @fbVariable@ is @a@
+* @fbSrc1@ corresponds to the @B@ comment
+* @fbSrc2@ corresponds to the @C@ comment
+* @fbAnnotation@ is @T@
+-}
+data FunctionBinding s a = FunctionBinding
+    { fbSrc0 :: Maybe s
+    , fbVariable :: Text
+    , fbSrc1 :: Maybe s
+    , fbSrc2 :: Maybe s
+    , fbAnnotation :: Expr s a
```

Comment:
So far the policy has been to largely avoid abbreviations like `fb`. Is there a reason for making an exception here?

Reply:
basically because I feel that `functionBindingAnnotation`, for instance, is quite large (`25` characters, to be precise). I have no problem with using the full name though.

Reply:
I agree that these would be rather long names, but I don't think that it's worth making an exception for.